internal/generator/tls.go

```diff
@@ -0,0 +1,39 @@
+package generator
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/tls"
+	"strings"
+)
+
+// TLSProfileInfo returns the minTLSVersion, ciphers as a comma-separated list given the available TLSSecurityProfile
+func (op Options) TLSProfileInfo(clfSpec *configv1.TLSSecurityProfile, outputSpec logging.OutputSpec) (string, string) {
```

Comment:
Another thing I'm not sure about, why make TLSProfileInfo a method on Options when all we use Options for is to get cluster-scope TLS profile.

Reply:
Wait, but after calling this, we turn around and 
op[generator.MinTLSVersion] = outMinTlsVersion
op[generator.Ciphers] = outCiphers
May be we should divorce TLSProfileInfo() from Options and use its return values to populate the two options ^^^.

Reply:
I have several motivations here:
* Profiles are or can be output specific so this is a convenient way to mod the generator behavior per output and consolidate the fall back logic
* Options are also the convenient way to pass around the feature gate to enable/disable the feature
* This puts the logic in a more singular place should we need to apply to the fluent impl

I like the idea of basically a factory that tells me how to configure the profile; it makes a decision like the following without polluting logic in several places with an if/else

```
		outMinTlsVersion, outCiphers := op.TLSProfileInfo(clfspec.TLSSecurityProfile, o)
		op[generator.MinTLSVersion] = outMinTlsVersion
		op[generator.Ciphers] = outCiphers
```

I'm open to an alternate impl if there is something cleaner